Fix links, tweak title of Geolocation in extensions guide (#6769)
GoogleChrome/developer.chrome.com: site/en/docs/extensions/mv3/geolocation/index.md

## Use geolocation in MV3 extensions

On the web, browsers safeguard users' geolocation data by showing a prompt asking them to grant that specific origin access to their location. The same permission model is not always appropriate for extensions.

<figure>
  {% Img src="image/DXqUldooyJOUnj3qXSYLHbUgUI93/nYRDFWbb280yEp90XDTI.png", alt="A screenshot of the permission prompt you see when a website requests access to the geolocation api", width="800", height="419" %}
  <figcaption>The geolocation permission prompt</figcaption>
</figure>

Permissions are not the only difference. As mentioned above, `navigator.geolocation` is a _DOM_ API, that is, something that is a part of the APIs that make up websites. As a result, it is not accessible inside worker contexts, like the [extension service worker][sw] that is the backbone of manifest v3 extensions. You can absolutely still use `geolocation`, though. There are just nuances with how and where you use it.

### Use geolocation in service workers

> There is no `navigator` object inside of service workers. It is only available inside contexts that have access to a page's `document` object. To get access inside of a service worker, use an [`Offscreen Document`]offscreen]. This extension API gives you access to an HTML file that you can bundle with your extension.

Reviewer: ```suggestion
There is no `navigator` object inside of service workers. It is only available inside contexts that have access to a page's `document` object. To get access inside of a service worker, use an [`Offscreen Document`][offscreen], which provides access to an HTML file that you can bundle with your extension.
```

> To get started, add `offscreen` to your manifest's `permissions` section.

Reviewer: ```suggestion
To get started, add `"offscreen"` to your manifest's `"permissions"` section.
```

<script src="offscreen.js"></script>
```

Save this file in the root of your project as `offscreen.html`.

> As mentioned, you need a script called `offscreen.js`. You'll also need to bundle this with your extension. It will be where you actually get geolocation information and send that information back to the service worker. You can send and receive messages between it and your service worker.

Reviewer: ```suggestion
As mentioned, you need a script called `offscreen.js`. You'll also need to bundle this with your extension. It will be the service worker's source of geolocation information. You can pass messages between it and your service worker.
```

const location = await getGeolocation()
```

### Use geolocation in a popup or side panel {: #geo-ext-pages}

> If you want to use geolocation within a [popup][popup] or [side panel][sidepanel], it is very straightforward. Popups and side panels are just web documents and therefore have access to the normal DOM APIs. You can access `navigator.geolocation` directly. The only difference from standard websites would be that you need to use the `manifest.json` `"permission"` field to request the `geolocation` permission. If you do not include the permission, you _will_ still have access to `navigator.geolocation`. However, any attempt to use it will cause an immediate error, the same as if the user rejected the request. You can see this in the [popup sample][popup-sample].

Reviewer: ```suggestion
Using geolocation within a [popup][popup] or [side panel][sidepanel] is very straightforward. Popups and side panels are just web documents and therefore have access to the normal DOM APIs. You can access `navigator.geolocation` directly. The only difference from standard websites is that you need to use the `manifest.json` `"permission"` field to request the `"geolocation"` permission. If you do not include the permission, you _will_ still have access to `navigator.geolocation`. However, any attempt to use it will cause an immediate error, the same as if the user rejected the request. You can see this in the [popup sample][popup-sample].
```

Reviewer: >If you do not include the permission, you _will_ still have access to `navigator.geolocation`. However, any attempt to use it will cause an immediate error

I'm guessing that unlike an extension API, calls to geolocation without permission don't throw and 'undefined' error. Who does this tidbit help and how does it help them? I think we could remove this text completely.

### Using geolocation in a content script {: #geo-content }

> Just like a popup, a [content script][content] has full access to the DOM API; however, you will go through the normal user permission flow. That means that adding `"geolocation"` to your `permissions` _will not_ automatically give you access to the users' geolocation information. You can see this in the [content script sample][contentscript-sample].

Reviewer: ```suggestion
Just like a popup, a [content script][content] has full access to the DOM API; however, users will go through the normal user permission flow. That means that adding `"geolocation"` to your `"permissions"` _will not_ automatically give you access to the users' geolocation information. You can see this in the [content script sample][contentscript-sample].
```
